source/PlayerInfo.cpp

```diff
@@ -2241,6 +2241,29 @@ void PlayerInfo::SelectNext()
 
 
 
+void PlayerInfo::DeselectAllSecondaries()
+{
+	selectedWeapons.clear();
+}
+
+
+
+void PlayerInfo::ToggleAnySecondary(const Outfit *outfit)
+{
+	if(!flagship)
+		return;
+
+	const auto it = selectedWeapons.find(outfit);
```

Comment:
You could be fancy and call `insert` here and then check the return value to see whether it was added or not (and if not then delete it).